src/store/auth/index.js

```diff
@@ -1,33 +1,80 @@
-import { auth } from 'firebase'
-import { firebaseMutations } from 'vuexfire'
+import firebase from '@/firebase'
+import { firebaseAction } from 'vuexfire'
+
+const auth = firebase.auth()
+const firestore = firebase.firestore()
+
+const getUserReference = user => firestore.collection('users').doc(user.uid)
 
 export default{
   namespaced: true,
   state: {
     // Add auth related state
-    user: null
+    user: null,
+    userData: null
   },
 
   getters: {
     // Add auth related getters
+    authenticated: state => state.user != null
   },
 
   actions: {
     // Add auth related actions
-    logout: () => auth().signOut()
+    setUser ({ commit, dispatch }, user) {
+      dispatch('fetchUserData', user)
+      dispatch('updateUserInformation', user)
```

Comment:
`fetchUserData` calls `getUserReference` and maps it to `userData`
The same thing happens inside `updateUserInformation`. 
Am I right ?
What we can do is:
commit(`fetchUserData`, user)    // This will wait until userData is updated
dispatch(`updateUserInformation`) // We can use the above userData here rather then making a call again.

Reply:
fetch is fetching information and updating is a one time process that happens when using the app for the first time

Reply:
Also, commit or a mutation is an atomic operation which should not be asynchronous, i.e., it should only change state in a predictable fashion. To do long running tasks, we use actions, so `commit(fetchUserData, user)` cannot wait

Reply:
Okaay @iamareebjamal Thanks! :+1: